src/services/pie-stat-getter.js

```diff
@@ -21,8 +21,8 @@ function PieStatGetter(model, params, opts) {
     const groupByFieldName = populateGroupByField
       ? params.group_by_field.replace(':', '.') : params.group_by_field;
 
-    return new P((resolve, reject) => {
-      const jsonQuery = queryBuilder.getQueryWithFiltersAndJoins(null, true);
+    return new P(async (resolve, reject) => {
+      const jsonQuery = await queryBuilder.getQueryWithFiltersAndJoins(null);
```

Comment:
pie chart stay very very long

Reply:
I can't really do a thing about that. This is due to the fact that if you do your pie chart on a million record it can't be fast. Mongo is pretty slow to join 2 collections, if you do it on a single collection the timing is alright

Reply:
I tried with filter on a belongs to it works perfectly fine (You need the next PR not this one 😉)

Reply:
Sure!